packages/plugins/src/hardhat/simulate.ts

```diff
@@ -275,10 +276,16 @@ export const simulateDeploymentSubtask = async (
     handleError: (e) => {
       throw e
     },
-    handleAlreadyExecutedDeployment: () => {
-      throw new Error(
-        'Deployment has already been executed. This is a bug. Please report it to the developers.'
-      )
+    handleAlreadyExecutedDeployment: async (deploymentContext) => {
+      executionCompleted = true
+      receipts = (
+        await fetchExecutionTransactionReceipts(
+          [],
+          deploymentContext.deployment.moduleAddress,
+          deploymentContext.deployment.deploymentConfig.merkleTree.root,
+          deploymentContext.provider
+        )
+      ).map(convertEthersTransactionReceipt)
     },
```

Comment:
Doing this here is probably a little overkill. Still, it keeps this function returning all the receipts in all situations, as is expected by the various places where this function is implemented. I think just adding this logic here is worth it for that purpose. This function already existed on the website, so I just moved it over into the mono repo.